Microsoft.Toolkit.Uwp.UI.Controls/AdvancedCollectionViewSource/AdvancedCollectionViewSource.Events.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml.Data;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls.AdvancedCollectionViewSource
+{
+    /// <summary>
+    /// A collection view source implementation that supports filtering, grouping, sorting and incremental loading
+    /// </summary>
+    public partial class AdvancedCollectionViewSource
+    {
+        /// <summary>
+        /// Currently selected item changing event
+        /// </summary>
+        /// <param name="e">event args</param>
+        protected virtual void OnCurrentChanging(CurrentChangingEventArgs e)
+        {
+            if (_deferCounter > 0)
+            {
+                return;
+            }
+
+            CurrentChanging?.Invoke(this, e);
```

Comment:
> Is this event needed? Wouldn't it be better to have oldValue and newValue in the event arguments of the CurrentChanged event?

The implementations are this way because of ICollectionView's requirements. (They are only used for the defer notifications part.)

> Also why virtual?

True. Removed virtual, and made it private.